views/map.js

```diff
@@ -147,6 +153,55 @@ class MapView extends Component {
 					'circle-color': VEHICLE_POINT_COLOR,
 				},
 			})
+
+			let popup = null
+			const showPopup = (e) => {
+				const feature = e.features[0]
+				if (!feature) return
+				const props = feature.properties
+
+				// Only show defined properties
+				const addProp = (label, value, code = true) => {
+					if (!value || value === 'null' || value === 'undefined') return ''
+					const displayValue = code ? `<code>${value}</code>` : value
+					return `${label}: ${displayValue}<br/>`
+				}
+
+				let vehicleSection = ''
+				vehicleSection += addProp('ID', props.vehicleId)
+				vehicleSection += addProp('Label', props.vehicleLabel)
+				vehicleSection += addProp('License Plate', props.vehicleLicensePlate)
+
+				let tripSection = ''
+				tripSection += addProp('ID', props.trip_id)
+				tripSection += addProp('Route', props.route_id)
+
+				const startDate = props.start_date.match(/^(\d{4})(\d{2})(\d{2})$/)
+				if (startDate && props.start_time) {
+					const start = new Date(Date.parse(`${startDate[1]}-${startDate[2]}-${startDate[3]}T${props.start_time}`))
+					// Default locale with short date/time
+					tripSection += addProp('Start', new Intl.DateTimeFormat(undefined, {
+						dateStyle: 'short',
+						timeStyle: 'short',
+					}).format(start), false)
+				}
+
+				let html = '<div>'
+				if (vehicleSection) {
+					html += '<b>Vehicle</b><br/>' + vehicleSection
+				}
+				if (tripSection) {
+					html += '<b>Trip</b><br/>' + tripSection
+				}
+				html += '</div>'
+
+				if (popup) popup.remove()
+				popup = new mapboxgl.Popup({closeButton: false, closeOnClick: false})
+					.setLngLat(e.lngLat)
```

Comment:
We should position the popup using the vehicle's map position, and also update it on a feed refresh.